### DISCUSSION TITLE
PL with Timm gives KeyError: 'model.model.conv1.0.bias when calling trainer.test

### DISCUSSION THREAD

**Discussion by MaxPolak97** [Lightning-AI/pytorch-lightning#15268] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,  I'm using Timm for creating Resnet model. For training it works fine but when calling ```trainer.test(light_model, datamodule, ckpt_path='best')``` I get the following error ```KeyError: 'model.model.conv1.0.bias'```. I have made three classes TransferLearningModel for loading any pre-trained model, DataModule for loading data and LightningClassifier model to do the actual training. So I checked how the model was defined and see that it goes wrong at .0 (see below). I didn't has this problem when using torchvision.models!
![image](https://user-images.githubusercontent.com/87903719/197470544-5523d97e-3c09-428b-85ff-2364c63199b4.png)

However, when I use [0] instead of .0, it works.
![image](https://user-images.githubusercontent.com/87903719/197469948-33944248-6462-4c48-bead-c1b107e8b33b.png)

Does anyone know how this problem occurred and can help me with solving this?

**Comment by akihironitta** [comment]
@MaxPolak97 Would you mind providing a minimal reproducible script and your env details to investigate this?
